docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,324 @@
+	CONFIG_OPTIMIZE_APP_RELOAD_TIME		Boolean		Optimizations for app reload time
+	---------------------------------------------------------------------------------------------
+```
+
+## Functions used and Enumeration for MPU regions in TizenRT
+
+- In case of no App-Binary Separation, the following 2 functions are used for configuring MPU regions for User Internal SRAM and User Flash.
+
+```c
+/****************************************************************************
+ * Name: mpu_user_flash
+ *
+ * Description:
+ *   Configure a region for user program flash
+ *
+ ****************************************************************************/
+static inline void mpu_userflash(uint32_t region, uintptr_t base, size_t size)
+{
+.
+.
+.
+}
+```
+```c
+/****************************************************************************
+ * Name: mpu_userintsram
+ *
+ * Description:
+ *   Configure a region as user internal SRAM
+ *
+ ****************************************************************************/
+static inline void mpu_userintsram(uint32_t region, uintptr_t base, size_t size)
+{
+.
+.
+.
+}
+```
+
+- In case of App-Binary Separation, below function is used to configure the user application SRAM MPU settings.
+
+```c
+
+/****************************************************************************
+ * Name: mpu_get_register_value
+ *
+ * Description:
+ *   Configure the user application SRAM mpu settings into the tcb variables
+ *
+ * Params:
+ *  regs     : pointer to array in which to store the configured values
+ *             If regs is NULL, then configure the mpu registers directly
+ *  region   : number of the region to be configured
+ *  base     : start address for the region
+ *  size     : size of the region in bytes
+ *  readonly : true indicates a readonly region
+ *  execute  : true indicates that the region has execute permission
+ ****************************************************************************/
+ 
+void mpu_get_register_value(uint32_t \*regs, uint32_t region, uintptr_t base, size_t size, uint8_t readonly, uint8_t execute)
+{
+.
+.
+.
+}
+```
+
+- Mapping of MPU regions to their respective memory segments has been done statically using below enumeration (in os/include/tinyara/mpu.h)
+
+```c
+enum MPU_REG_NUM {
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	MPU_REG_NUM_COM_LIB_TXT,	/* Common Binary */
+	MPU_REG_NUM_COM_LIB_RO,
+	MPU_REG_NUM_COM_LIB_DATA,
+#else
+	MPU_REG_NUM_COM_LIB,
+#endif
+#endif
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	MPU_REG_NUM_APP_TXT,		/* Apps */
+	MPU_REG_NUM_APP_RO,
+	MPU_REG_NUM_APP_DATA,
+#else
+	MPU_REG_NUM_APP,
+#endif
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+	MPU_REG_NUM_STK,		/* Stack */
+#endif
+#else
+	MPU_REG_NUM_UFLASH,
+	MPU_REG_NUM_URAM,
+#endif
+	MPU_REG_NUM_MAX
+};
+```
+
+## How to Control MPU regions in TizenRT
+### Add a new MPU region
+1. Add your MPU region in enum MPU_REG_NUM in mpu.h file.
```

Comment:
You need to mention here that if we require one-time configuration, then any of the mpu_xxxx api from mpu.h can be used to configure the region. The next point should say that, if repeated configuration is required, then use the two step process.

Reply:
done